File: VB/CustomFormats/Code/Services.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports DevExpress.XtraScheduler.Services
Imports DevExpress.XtraScheduler.Drawing
Imports DevExpress.Web.ASPxScheduler.Services
Imports DevExpress.XtraScheduler

Public Class CustomTimeRulerFormatStringService
	Inherits TimeRulerFormatStringServiceWrapper
	Public Sub New(ByVal service As ITimeRulerFormatStringService)
		MyBase.New(service)
	End Sub

	Public Overrides Function GetHalfDayHourFormat(ByVal ruler As TimeRuler) As String
		If ruler.UseClientTimeZone Then
			Return "'HalfDayHour'"
		Else
			Return "HH:mm"
		End If
	End Function
	Public Overrides Function GetHourFormat(ByVal ruler As TimeRuler) As String
		If ruler.UseClientTimeZone Then
			Return "HH - 'Hour'"
		Else
			Return "HH:mm"
		End If
	End Function
	Public Overrides Function GetHourOnlyFormat(ByVal ruler As TimeRuler) As String
		If ruler.UseClientTimeZone Then
			Return "HH -> 'HourOnly'"
		Else
			Return "HH"
		End If
	End Function
	Public Overrides Function GetMinutesOnlyFormat(ByVal ruler As TimeRuler) As String
		If ruler.UseClientTimeZone Then
			Return "mm - 'MinutesOnly'"
		Else
			Return "mm"
		End If
	End Function
	Public Overrides Function GetTimeDesignatorOnlyFormat(ByVal ruler As TimeRuler) As String
		If ruler.UseClientTimeZone Then
			Return "tt - 'TimeDesignatorOnly'"
		Else
			Return "mm"
		End If
	End Function
End Class

Public Class CustomAppointmentFormatStringService
	Inherits AppointmentFormatStringServiceWrapper
	Public Sub New(ByVal service As IAppointmentFormatStringService)
		MyBase.New(service)
	End Sub
	Public Overrides Function GetVerticalAppointmentStartFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
		Return "{0: HHmm:ss}"
	End Function
	Public Overrides Function GetVerticalAppointmentEndFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
		Return "{0: HHmm:ss}"
	End Function
	Public Overrides Function GetHorizontalAppointmentEndFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
		Return "{0: HHmm}"
	End Function
	Public Overrides Function GetHorizontalAppointmentStartFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
		Return "{0: HHmm}"
	End Function
	Public Overrides Function GetContinueItemStartFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
		Return "{0: HHmm MMM dd}"
	End Function
	Public Overrides Function GetContinueItemEndFormat(ByVal aptViewInfo As IAppointmentViewInfo) As String
		Return "{0: HHmm MMM dd}"
	End Function
End Class
Public Class CustomHeaderCaptionService
	Inherits HeaderCaptionServiceWrapper
	Public Sub New(ByVal service As IHeaderCaptionService)
		MyBase.New(service)
	End Sub

	Public Overrides Function GetDayColumnHeaderCaption(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDateHeader) As String
		Dim [date] As DateTime = header.Interval.Start.Date
		If [date].Month = 1 AndAlso [date].Day = 1 Then
			Return "Happy New Year!"
		Else
			Return MyBase.GetDayColumnHeaderCaption(header)
		End If
	End Function


	Public Overrides Function GetDayOfWeekHeaderCaption(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDayOfWeekHeader) As String
		Dim [date] As DayOfWeek = header.DayOfWeek
		If [date] = DayOfWeek.Friday Then
			Return "TGIF!"
		Else
			Return MyBase.GetDayOfWeekHeaderCaption(header)
		End If
	End Function

	Public Overrides Function GetHorizontalWeekCellHeaderCaption(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader) As String
		Dim [date] As DateTime = header.Interval.Start.Date
		If [date].Month = 1 AndAlso [date].Day = 1 Then
			Return "HorizontalWeekCellHeader"
		Else
			Return [date].ToString("D")
		End If

	End Function

	Public Overrides Function GetTimeScaleHeaderCaption(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebTimeScaleHeader) As String
		Dim [date] As DateTime = header.Interval.Start.Date
		If [date].Month = 1 AndAlso [date].Day = 1 Then
			Return "TimeScaleHeader"
		Else
			Return MyBase.GetTimeScaleHeaderCaption(header)
		End If

	End Function

	Public Overrides Function GetVerticalWeekCellHeaderCaption(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader) As String
		Dim [date] As DateTime = header.Interval.Start.Date
		If [date].Month = 1 AndAlso [date].Day = 1 Then
			Return "VerticalWeekCellHeader"
		Else
			Return MyBase.GetVerticalWeekCellHeaderCaption(header)
		End If

	End Function
End Class

Public Class CustomHeaderToolTipService
	Inherits HeaderToolTipServiceWrapper
	Public Sub New(ByVal service As IHeaderToolTipService)
		MyBase.New(service)
	End Sub

	Public Overrides Function GetDayColumnHeaderToolTip(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDateHeader) As String
		Dim [date] As DateTime = header.Interval.Start.Date
		If [date].Month = 1 AndAlso [date].Day = 1 Then
			Return "Let's celebrate!"
		Else
			Return MyBase.GetDayColumnHeaderToolTip(header)
		End If
	End Function

	Public Overrides Function GetDayOfWeekHeaderToolTip(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDayOfWeekHeader) As String
		If header.DayOfWeek = DayOfWeek.Friday Then
			Return "TGIF!"
		Else
			Return MyBase.GetDayOfWeekHeaderToolTip(header)
		End If
	End Function

	Public Overrides Function GetTimeScaleHeaderToolTip(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebTimeScaleHeader) As String
		If header.Interval.Contains(New DateTime(2009, 1, 1)) Then
			Return "Let's celebrate!"
		Else
			Return MyBase.GetTimeScaleHeaderToolTip(header)
		End If
	End Function
	Public Overrides Function GetHorizontalWeekCellHeaderToolTip(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader) As String
		If header.Interval.IntersectsWith(New TimeInterval(New DateTime(2008, 12, 24), New DateTime(2009, 1, 4))) Then
			Return "Merry Christmas and Happy New Year!"
		Else
		Return MyBase.GetHorizontalWeekCellHeaderToolTip(header)
		End If
	End Function
	Public Overrides Function GetVerticalWeekCellHeaderToolTip(ByVal header As DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader) As String
		If header.Interval.IntersectsWith(New TimeInterval(New DateTime(2008, 12, 24), New DateTime(2009, 1, 4))) Then
			Return "Merry Christmas and Happy New Year!"
		Else
		Return MyBase.GetVerticalWeekCellHeaderToolTip(header)
		End If
	End Function
End Class

File: CS/CustomFormats/Code/Services.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using DevExpress.XtraScheduler.Services;
using DevExpress.XtraScheduler.Drawing;
using DevExpress.Web.ASPxScheduler.Services;
using DevExpress.XtraScheduler;

public class CustomTimeRulerFormatStringService : TimeRulerFormatStringServiceWrapper {
    public CustomTimeRulerFormatStringService(ITimeRulerFormatStringService service)
        : base(service) {
    }

    public override string GetHalfDayHourFormat(TimeRuler ruler) {
        return ruler.UseClientTimeZone ? "'HalfDayHour'" : "HH:mm";
    }
    public override string GetHourFormat(TimeRuler ruler) {
        return ruler.UseClientTimeZone ? "HH - 'Hour'" : "HH:mm";
    }
    public override string GetHourOnlyFormat(TimeRuler ruler) {
        return ruler.UseClientTimeZone ? "HH -> 'HourOnly'" : "HH";
    }
    public override string GetMinutesOnlyFormat(TimeRuler ruler) {
        return ruler.UseClientTimeZone ? "mm - 'MinutesOnly'" : "mm";
    }
    public override string GetTimeDesignatorOnlyFormat(TimeRuler ruler) {
        return ruler.UseClientTimeZone ? "tt - 'TimeDesignatorOnly'" : "mm";
    }
}

public class CustomAppointmentFormatStringService : AppointmentFormatStringServiceWrapper {
    public CustomAppointmentFormatStringService(IAppointmentFormatStringService service)
        : base(service) {
    }
    public override string GetVerticalAppointmentStartFormat(IAppointmentViewInfo aptViewInfo) {
        return "{0: HHmm:ss}";
    }
    public override string GetVerticalAppointmentEndFormat(IAppointmentViewInfo aptViewInfo) {
        return "{0: HHmm:ss}";
    }
    public override string GetHorizontalAppointmentEndFormat(IAppointmentViewInfo aptViewInfo) {
        return "{0: HHmm}";
    }
    public override string GetHorizontalAppointmentStartFormat(IAppointmentViewInfo aptViewInfo) {
        return "{0: HHmm}";
    }
    public override string GetContinueItemStartFormat(IAppointmentViewInfo aptViewInfo) {
        return "{0: HHmm MMM dd}";
    }
    public override string GetContinueItemEndFormat(IAppointmentViewInfo aptViewInfo) {
        return "{0: HHmm MMM dd}";
    }
}
public class CustomHeaderCaptionService : HeaderCaptionServiceWrapper {
    public CustomHeaderCaptionService(IHeaderCaptionService service)
        : base(service) {
    }

    public override string GetDayColumnHeaderCaption(DevExpress.Web.ASPxScheduler.Rendering.WebDateHeader header) {
        DateTime date = header.Interval.Start.Date;
        if (date.Month == 1 && date.Day == 1)
            return "Happy New Year!";
        else
            return base.GetDayColumnHeaderCaption(header);
    }


    public override string GetDayOfWeekHeaderCaption(DevExpress.Web.ASPxScheduler.Rendering.WebDayOfWeekHeader header) {
        DayOfWeek date = header.DayOfWeek;
        if (date == DayOfWeek.Friday)
            return "TGIF!";
        else
            return base.GetDayOfWeekHeaderCaption(header);
    }

    public override string GetHorizontalWeekCellHeaderCaption(DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader header) {
        DateTime date = header.Interval.Start.Date;
        if (date.Month == 1 && date.Day == 1)
            return "HorizontalWeekCellHeader";
        else
            return date.ToString("D");

    }
   
    public override string GetTimeScaleHeaderCaption(DevExpress.Web.ASPxScheduler.Rendering.WebTimeScaleHeader header) {
        DateTime date = header.Interval.Start.Date;
        if (date.Month == 1 && date.Day == 1)
            return "TimeScaleHeader";
        else
            return base.GetTimeScaleHeaderCaption(header);

    }

    public override string GetVerticalWeekCellHeaderCaption(DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader header) {
        DateTime date = header.Interval.Start.Date;
        if (date.Month == 1 && date.Day == 1)
            return "VerticalWeekCellHeader";
        else
            return base.GetVerticalWeekCellHeaderCaption(header);

    }
}

public class CustomHeaderToolTipService : HeaderToolTipServiceWrapper {
    public CustomHeaderToolTipService(IHeaderToolTipService service)
        : base(service) {
    }
  
    public override string GetDayColumnHeaderToolTip(DevExpress.Web.ASPxScheduler.Rendering.WebDateHeader header) {
        DateTime date = header.Interval.Start.Date;
        if (date.Month == 1 && date.Day == 1)
            return "Let's celebrate!";
        else
            return base.GetDayColumnHeaderToolTip(header);
    }
    
    public override string GetDayOfWeekHeaderToolTip(DevExpress.Web.ASPxScheduler.Rendering.WebDayOfWeekHeader header) {
        if (header.DayOfWeek == DayOfWeek.Friday)
            return "TGIF!";
        else
            return base.GetDayOfWeekHeaderToolTip(header);
    }

    public override string GetTimeScaleHeaderToolTip(DevExpress.Web.ASPxScheduler.Rendering.WebTimeScaleHeader header) {
        if (header.Interval.Contains(new DateTime(2009, 1, 1)))
            return "Let's celebrate!";
        else
            return base.GetTimeScaleHeaderToolTip(header);
    }
    public override string GetHorizontalWeekCellHeaderToolTip(DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader header) {
        if (header.Interval.IntersectsWith(new TimeInterval(new DateTime(2008, 12, 24), new DateTime(2009, 1, 4))))
            return "Merry Christmas and Happy New Year!";
        else
        return base.GetHorizontalWeekCellHeaderToolTip(header);
    }
    public override string GetVerticalWeekCellHeaderToolTip(DevExpress.Web.ASPxScheduler.Rendering.WebDateCellHeader header) {
        if (header.Interval.IntersectsWith(new TimeInterval(new DateTime(2008, 12, 24), new DateTime(2009, 1, 4))))
            return "Merry Christmas and Happy New Year!";
        else
        return base.GetVerticalWeekCellHeaderToolTip(header);
    }
}